### PR TITLE
Allow react 16 as a peer dependency

### PR DESCRIPTION
**Pull request by gpeal**
Ran tests with react 15 and react 16 and all passed.
This is necessary to use with react native >= 0.43 (latest is 0.47-rc).
@Spy-Seth @vvo

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,7 +66,7 @@
     "react-test-renderer": "15.6.1"
   },
   "peerDependencies": {
-    "react": "^0.14.8 || ^15.0.1"
+    "react": "^0.14.8 || ^15.0.1 || ^16.0.0 || ^16.0.0-beta || ^16.0.0-alpha"
```
Comment (Spy-Seth) [inline package.json]: Could you also allow the release candidate?

**Comment by Spy-Seth** [comment]
#127 is green! 🎉

**Comment by gpeal** [comment]
@Spy-Seth They're green :)

File: package.json (hunk 1 above)
Comment (Spy-Seth) [inline package.json]: This is still missing :)

**Comment by gpeal** [comment]
@Spy-Seth Done!

**Comment by vvo** [comment]
Hey hi @gpeal, thanks for the contribution. How are you using this library today? What's your usecase?

**Comment by gpeal** [comment]
We (Airbnb) use chai-enzyme (which has a [dep](https://github.com/producthunt/chai-enzyme/blob/master/package.json#L88) of react-element-to-jsx-string) and are upgrading to React Native 0.46 which has a dep of react-16.0.0-alpha.12 so we need to go and update all of our deps and their transitive deps.

**Pull request merged**